src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -187,5 +187,45 @@ public static bool StartsWith<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> valu
             int valueLength = value.Length;
             return valueLength <= span.Length && SpanHelpers.SequenceEqual(ref span.DangerousGetPinnableReference(), ref value.DangerousGetPinnableReference(), valueLength);
         }
+
+        /// <summary>
+        /// Determines whether the specified sequence appears at the start of the span.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ReadOnlySpan<char> AsSpan(this string text)
+        {
+            if (text == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
+
+            int textLength = text.Length;
+
+            if (textLength == 0) return ReadOnlySpan<char>.Empty;
+
+            unsafe
+            {
+                fixed (char* charPointer = text)
```

Comment:
In the email conversation, you have said that you are going to add this to corefxlab experimental. Why are you adding it here now? This is like 2x slower than it needs to be for the fast span. If you want to add the implementation to corefx, it should be the proper performant implementation.